### PR TITLE
[libsocialcache] Added support for OneDrive images. Fixes MER#1333

### PR DESCRIPTION
**Pull request by antseppa**

### CODE REVIEW

**Comment by antseppa** [comment]
Updated: added keyprovider support for onedrive.

**Comment by antseppa** [comment]
Updated: modified SocialImageCache so that it is possible to cache images also based on imageId (earlier only based on url). OneDrive image download urls are temporary, so that was needed.

File: tests/tst_onedriveimage/main.cpp [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+        QCOMPARE(users[0]->updatedTime(), time1);
+        QCOMPARE(users[0]->userName(), QLatin1String("c"));
+        QCOMPARE(users[0]->count(), 0);
+
+        QCOMPARE(users[1]->userId(), QLatin1String("d"));
+        QCOMPARE(users[1]->updatedTime(), time2);
+        QCOMPARE(users[1]->userName(), QLatin1String("f"));
+        QCOMPARE(users[1]->count(), 0);
+
+/*        OneDriveImageCacheModel model;
```
Comment (chriadam) [inline tests/tst_onedriveimage/main.cpp]: why comment out this block?

Comment (antseppa) [inline tests/tst_onedriveimage/main.cpp]: Accidently, actually :) It passes fine, was doing some debugging during development and forgot it like that. Will fix.

**Comment by antseppa** [comment]
Updated: uncommented a test case and changed sorting order to descending also in album view.

**Comment by antseppa** [comment]
Updated: added description field to images and accountId field to users.

File: src/lib/socialimagesdatabase.cpp [hunk 2]
```diff
@@ -498,7 +552,8 @@ bool SocialImagesDatabase::createTables(QSqlDatabase database) const
                   "imageUrl TEXT,"
                   "imageFile TEXT,"
                   "createdTime INTEGER,"
-                  "expires INTEGER)");
+                  "expires INTEGER,"
+                  "imageId STRING);");
```
Comment (rainisto) [inline src/lib/socialimagesdatabase.cpp]: );"); -> )");

**Comment by chriadam** [comment]
Closing this one as I've cherry-picked the appropriate commit into master from cloud branch after massaging history a bit for neatness.

**Pull request closed without merging**